src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -27,9 +33,18 @@ public BuildJobExtension()
                 displayName: StringUtil.Loc("GetSources"),
                 enabled: true,
                 @finally: false);
+
+            FinallyStep = new JobExtensionRunner(
+                runAsync: PrepareAsync,
+                alwaysRun: false,
+                continueOnError: false,
+                critical: false,
+                displayName: StringUtil.Loc("CleanUp"),
```

Comment:
how should we name the step? the step will remove embedded credential if the repository is git.